### DISCUSSION TITLE
支不支持客户端支不支持局域网，类似于 allow-lan 的功能？

### DISCUSSION THREAD

**Discussion by wandersofb** [apernet/hysteria#776] [General]
支不支持客户端支不支持局域网，类似于 allow-lan 的功能？

**Comment by haruue** [comment]
hysteria 客户端的 [socks5 / http 入站](https://hysteria.network/zh/docs/advanced/Full-Client-Config/#_6) 监听这个设备的局域网地址， 就能允许局域网设备通过配置 http / socks5 使用代理。

例如假设你把 hysteria 客户端跑在你局域网中的某一台设备上， 这个设备的 IP 地址是 `192.0.2.233` ， 那你可以像这样编写配置文件。

```
socks5:
  listen: 192.0.2.233:1080
```

这样局域网中的其它设备就能通过配置 socks5 代理为 `192.0.2.233:1080` 来使用 hysteria 客户端提供的代理。 即为「允许局域网设备连接」。

此外配置成 `listen: :1080` 以监听任意地址也是可行的， 但是要小心不要把无认证的 http / socks5 代理开在公网上。
